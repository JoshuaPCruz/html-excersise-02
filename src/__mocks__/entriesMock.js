import { entries } from '../manifest/manifest.json';

export default entries;