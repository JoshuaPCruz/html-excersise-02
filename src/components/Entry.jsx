import React from 'react';
import '../styles/components/Entry.css';

const Entry = ({ date, author, comments, title, body }) => {
  return (
    <>
      <div className="Entry">
        <div className="Entry-metadata">
          <p>El {date}</p>
          <p>
            Por <a href="">{author}</a>
          </p>
          <small>{comments} comentarios</small>
        </div>
        <div className="Entry-content">
          <h1>{title}</h1>
          <p>
            {body}
          </p>
          <a href="">Leer m√°s</a>
        </div>
      </div>
    </>
  );
};

export default Entry;