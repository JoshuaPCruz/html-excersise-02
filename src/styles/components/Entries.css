.Entries {
  border-radius: 10px;
  padding: 15px;
  margin-bottom: 20px;
  background-color: var(--color-secondary-white);
}

.Entries hr {
  margin-bottom: 20px;
}