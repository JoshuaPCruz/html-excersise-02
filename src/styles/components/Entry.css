.Entry {
  margin-bottom: 20px;
  display: flex;
}

.Entry-metadata {
  width: 50%;
}

.Entry-metadata p {
  margin: 5px 0;
}

.Entry-metadata a {
  color: var(--color-primary-red);
  text-decoration: none;
}

.Entry-metadata small {
  font-style: italic;
}

.Entry-content h1 {
  margin-bottom: 10px;
  color: var(--color-primary-red);
  font-weight: bold;
  font-size: 1.5em;
}

.Entry-content p {
  margin: 10px 0 30px;
}

.Entry-content a {
  color: var(--color-primary-red);
  text-decoration: none;
}