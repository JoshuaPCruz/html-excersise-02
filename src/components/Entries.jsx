import React, { Fragment } from 'react';
import '../styles/components/Entries.css';
import Entry from './Entry';

const Entries = ({ entries }) => {
  return (
    <div className="Entries">
      {
        entries.map((entry, index) => {
          const isLast = index !== entries.length - 1;
          return <Fragment key={index}>
            <Entry {...entry} /> {isLast && <hr />}
          </Fragment>
        })
      }
    </div>
  );
};

export default Entries;