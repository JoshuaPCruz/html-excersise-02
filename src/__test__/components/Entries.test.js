import React from 'react';
import { shallow } from 'enzyme';
import { create } from 'react-test-renderer';
import Entries from '../../components/Entries';
import Entry from '../../components/Entry';
import entriesMock from '../../__mocks__/entriesMock';

describe('<Entries />', () => {
  const entries = shallow(<Entries entries={entriesMock} />);
  test('Render del componente Entries', () => {
    expect(entries.length).toEqual(1);
  });

  test('Render de la primera entry', () => {
    expect(entries.contains(<Entry {...entriesMock[0]} />)).toBeTruthy();
  });
});

describe('Entries Snapshot', () => {
  test('Comprobar la UI del componente Entries', () => {
    const entries = create(<Entries entries={entriesMock} />);
    expect(entries.toJSON()).toMatchSnapshot();
  });
});