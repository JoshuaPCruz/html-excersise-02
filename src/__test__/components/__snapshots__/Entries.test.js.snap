// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Entries Snapshot Comprobar la UI del componente Entries 1`] = `
<div
  className="Entries"
>
  <div
    className="Entry"
  >
    <div
      className="Entry-metadata"
    >
      <p>
        El 
        29 de Octubre de 2012
      </p>
      <p>
        Por 
        <a
          href=""
        >
          Ernesto G Bustamante
        </a>
      </p>
      <small>
        3
         comentarios
      </small>
    </div>
    <div
      className="Entry-content"
    >
      <h1>
        Las bondades de HTML5
      </h1>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Amet, totam nemo esse, aspernatur maxime iste dolore quod, impedit eaque corrupti deserunt voluptatum itaque eum porro consequuntur ad dolorum reiciendis. Dolorem?
      </p>
      <a
        href=""
      >
        Leer más
      </a>
    </div>
  </div>
   
  <hr />
  <div
    className="Entry"
  >
    <div
      className="Entry-metadata"
    >
      <p>
        El 
        El 28 de Octubre de 2012
      </p>
      <p>
        Por 
        <a
          href=""
        >
          Ernesto G Bustamante
        </a>
      </p>
      <small>
        21
         comentarios
      </small>
    </div>
    <div
      className="Entry-content"
    >
      <h1>
        HTML5 en dispositivos móviles
      </h1>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Amet, totam nemo esse, aspernatur maxime iste dolore quod, impedit eaque corrupti deserunt voluptatum itaque eum porro consequuntur ad dolorum reiciendis. Dolorem?
      </p>
      <a
        href=""
      >
        Leer más
      </a>
    </div>
  </div>
   
  <hr />
  <div
    className="Entry"
  >
    <div
      className="Entry-metadata"
    >
      <p>
        El 
        27 de Octubre de 2012
      </p>
      <p>
        Por 
        <a
          href=""
        >
          Ernesto G Bustamante
        </a>
      </p>
      <small>
        6
         comentarios
      </small>
    </div>
    <div
      className="Entry-content"
    >
      <h1>
        Internet explorer es una basura
      </h1>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Amet, totam nemo esse, aspernatur maxime iste dolore quod, impedit eaque corrupti deserunt voluptatum itaque eum porro consequuntur ad dolorum reiciendis. Dolorem?
      </p>
      <a
        href=""
      >
        Leer más
      </a>
    </div>
  </div>
   
  <hr />
  <div
    className="Entry"
  >
    <div
      className="Entry-metadata"
    >
      <p>
        El 
        27 de Octubre de 2012
      </p>
      <p>
        Por 
        <a
          href=""
        >
          Ernesto G Bustamante
        </a>
      </p>
      <small>
        2
         comentarios
      </small>
    </div>
    <div
      className="Entry-content"
    >
      <h1>
        Diseño web en plantillas retina
      </h1>
      <p>
        Lorem, ipsum dolor sit amet consectetur adipisicing elit. Amet, totam nemo esse, aspernatur maxime iste dolore quod, impedit eaque corrupti deserunt voluptatum itaque eum porro consequuntur ad dolorum reiciendis. Dolorem?
      </p>
      <a
        href=""
      >
        Leer más
      </a>
    </div>
  </div>
   
</div>
`;
